nixos/modules/services/backup/borgbackup.nix

```diff
@@ -19,7 +19,8 @@ let
     concatStringsSep " "
       (mapAttrsToList (x: y: "--keep-${x}=${toString y}") cfg.prune.keep);
 
-  mkBackupScript = cfg: ''
+  mkBackupScript = name: cfg: pkgs.writeShellScript "${name}-script" (''
+    set -e
```

Comment:
Is this a leftover of debugging?

Reply:
Are you referring to `set -x`, which is used to e.g. echo commands as they are executed? [`set -e`](https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html) instead has a behavior effect on the script:

> -e
>
>    Exit immediately if a pipeline (see [Pipelines](https://www.gnu.org/software/bash/manual/html_node/Pipelines.html)), which may consist of a single simple command (see [Simple Commands](https://www.gnu.org/software/bash/manual/html_node/Simple-Commands.html)), a list (see [Lists of Commands](https://www.gnu.org/software/bash/manual/html_node/Lists.html)), or a compound command (see [Compound Commands](https://www.gnu.org/software/bash/manual/html_node/Compound-Commands.html)) returns a non-zero status.

I added this so that we have the same behavior (i.e. refactoring), although I did not check whether it makes a difference in every case. `set -e` is not provided by `pkgs.writeShellScript` by default, whereas `set -e` is provided by default by `systemd.services.<name>.script` [via `makeJobScript`](https://github.com/NixOS/nixpkgs/blob/596a8e828c5dfa504f91918d0fa4152db3ab5502/nixos/lib/systemd-lib.nix#L263). So to make it so that the backup script continues to run with `set -e`, I needed to add this.

Reply:
Right, sorry for making you explain all of this :'( (but thank you for the amazing comment!)